core/src/main/java/com/linecorp/armeria/client/HttpResponseWrapper.java

```diff
@@ -50,6 +51,10 @@ class HttpResponseWrapper implements StreamWriter<HttpObject> {
 
     private static final Logger logger = LoggerFactory.getLogger(HttpResponseWrapper.class);
 
+    // This handler is used only when the response requires changing the state of the next request.
+    // Otherwise, it should be null.
```

Comment:
This is not true since it is actually always non-null except when we make an HTTP2 upgrade request

```suggestion
```